docs/event-processing/idempotent-writer.md

```diff
@@ -0,0 +1,36 @@
+---
+seo:
+  title: Idempotent Writer
+  description: An Idempotent Writer produces an Event to an Event Streaming Platform exactly once.
+---
+
+# Idempotent Writer
+A writer produces [Events](../event/event.md) that are written into an [Event Stream](../event-stream/event-stream.md), and under stable conditions, each Event is recorded only once.
+However, in the case of an operational failure or a brief network outage, an [Event Source](../event-source/event-source.md) may need to retry writes. This may result in multiple copies of the same Event ending up in the Event Stream, as the first write may have actually succeeded even though the producer did not receive the acknowledgement from the [Event Streaming Platform](../event-stream/event-streaming-platform.md). This type of duplication is a common failure scenario in practice and one of the perils of distributed systems.
+
+## Problem
+How can an [Event Streaming Platform](../event-stream/event-streaming-platform.md) ensure that an Event Source does not write the same Event more than once?
+
+## Solution
+![idempotent-writer](../img/idempotent-writer.svg)
+
+Generally speaking, this can be addressed by native support for idempotent clients.
+This means that a writer may try to produce an Event more than once, but the Event Streaming Platform detects and discards duplicate write attempts for the same Event.
+
+## Implementation
```

Comment:
Sorry, I forgot this in my review yesterday:  Please also add ksqlDB and Kafka Streams. This can be very short, but we should tell the reader how to enable this via EOS for those two as well.